synapse/rest/client/room.py

```diff
@@ -587,14 +620,23 @@ class RoomMessageListRestServlet(RestServlet):
     def __init__(self, hs: "HomeServer"):
         super().__init__()
         self._hs = hs
+        self.clock = hs.get_clock()
         self.pagination_handler = hs.get_pagination_handler()
         self.auth = hs.get_auth()
         self.store = hs.get_datastores().main
 
-    @messsages_response_timer.time()
     async def on_GET(
         self, request: SynapseRequest, room_id: str
     ) -> Tuple[int, JsonDict]:
+        processing_start_time = self.clock.time_msec()
+        # Fire and forget and hope that we get a result by the end.
+        #
+        # Ideally we'd use `Deferred.fromCoroutine()` here, to save on redundant
+        # type-checking, but we'd need Twisted >= 21.2.
```

Comment:
Comment copied from: https://github.com/matrix-org/synapse/blob/49d04e43dfa0551f52c1f1872b6f311efa756ca8/synapse/util/async_helpers.py#L760-L761

Because the Twisted docs mention:

> The [`ensureDeferred`](https://docs.twistedmatrix.com/en/stable/api/twisted.internet.defer.html#ensureDeferred) function also provides a way to convert a coroutine to a Deferred, but it’s interface is more type-ambiguous; `Deferred.fromCoroutine` is meant to replace it.
>
> *-- https://docs.twistedmatrix.com/en/stable/core/howto/defer-intro.html#coroutines-with-async-await*